### DISCUSSION TITLE
How to conveniently go to a plugins config file using LSP's `gd`?

### DISCUSSION THREAD

**Discussion by johnmathews** [wbthomason/packer.nvim#791] [Q&A] [answered]
I store the config for many plugins in separate Lua files. For each plugin,  the `config` param points to a function that returns the config file path as a string. I'd like to open the file at this location using LSP's "goto definition", instead of manually navigating to them or opening telescope. 

Because the config parameter requires either a string or function that returns a string, but not the `require` builtin (if that's what to call it), using `vim.lsp.buf.definition()` doesn't work. 

What's a good workaround for this?

Example:
```
local function get_config(name)
  return string.format('require("plugins/%s")', name)
end
```

```
use({ "goolord/alpha-nvim", config = get_config("alpha") })
```

**Comment by johnmathews** [comment, marked as answer]
The best answer I've seen is to use an anonymous function

```
use({
  "goolord/alpha-nvim",
    config = function()
      require("plugins.alpha")
    end,
})
```

**Comment by filipekiss** [comment]
You can also use Vim's `set path` command to add `plugins/` to the list and use `gf` instead of `gd` (`g`o to `f`ile)

Say your plugin is located in `lua/user/plugins/alpha.lua`. You can do `set path+=lua/user/plugins/*` and when you press `gf` in the `alpha` string, it will send you to `lua/user/plugins/alpha.lua`.
